.github/workflows/ci.yaml

```diff
@@ -0,0 +1,332 @@
+  olm:
+    runs-on: ubuntu-20.04
+    needs:
+      - lint
+      - unit
+    defaults:
+      run:
+        shell: bash
+        working-directory: metallboperator
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Setup
+      uses: ./.github/workflows/composite/setup
+  
+    - name: Verify release bundle manifests
+      run: |
+        make bundle-release
+        git diff --exit-code -I'^    createdAt: ' bundle
+
+    - name: Create and set up K8s Kind Cluster
+      run: |
+        ./hack/kind-cluster-with-registry.sh
+        make deploy-olm
+
+    - name: Build bundle image
+      run: |
+        make build-and-push-bundle-images REPO=localhost:5000
+
+    - name: Deploy Metal LB Operator with OLM
+      run: |
+        make deploy-with-olm REPO=localhost:5000
+    - name: E2E Tests
+      run: |
+        export KUBECONFIG=${HOME}/.kube/config
+        make test-validation
+        make test-e2e
+    - name: Collect Logs
+      uses: ./.github/workflows/composite/collectlogs
+  upgrade_version:
+    runs-on: ubuntu-20.04
+    env:
+      built_image_prev_release: "metallb-operator-prev-rel:ci" # Arbitrary name
+      built_image: "metallb-operator:ci" # Arbitrary name
+    name: Go 1.17 -> Go 1.20
+    defaults:
+      run:
+        shell: bash
+        working-directory: metallboperator
+    needs:
+      - lint
+      - unit
+    steps:
+    - name: Cancel Previous Runs
+      uses: styfle/cancel-workflow-action@0.9.1
+      with:
+        access_token: ${{ github.token }}
+    - name: Checkout Previous Release Metal LB Operator
+      uses: actions/checkout@v2
+      with:
+        path: metallboperator
+        ref: v0.12 # previous release version
+        fetch-depth: 0 # Fetch all history for all tags and branches
+
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '1.17'
+
+    - name: Build image
+      run: |
+        IMG=${built_image_prev_release} make docker-build
+    - name: Create K8s Kind Cluster
+      run: |
+        ./hack/kind-cluster-without-registry.sh
+        kind load docker-image ${built_image_prev_release}
+    - name: Deploy Previous Release Metal LB Operator
+      run: |
+        make deploy-cert-manager
+        IMG=${built_image_prev_release} KUSTOMIZE_DEPLOY_DIR="config/kind-ci/" ENABLE_OPERATOR_WEBHOOK="true" make deploy
+    - name: E2E Tests
+      run: |
+        export KUBECONFIG=${HOME}/.kube/config
+        make test-validation
+        make test-e2e
+    - name: Archive E2E Tests logs
+      if: ${{ failure() }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: test_e2e_logs
+        path: /tmp/test_e2e_logs/
+    - name: Export kind logs
+      if: ${{ failure() }}
+      run: |
+        kind export logs /tmp/kind_logs
+    - name: Archive kind logs
+      if: ${{ failure() }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: kind_logs
+        path: /tmp/kind_logs
```

Comment:
Let's use the collectlogs composite here also

Reply:
I did that but it failed and once I used without collectlogs composite it works. 
I will try again to use in collectlogs composite.

Reply:
we decided on keeping it like this here, and using the collectlogs composite in the second time.

Reply:
Finally , I added **if: ${{ failure() }}** and it works @liornoy 
Thanks to @oribon  on the commenting.